### ISSUE TITLE
changelog/release for 0.32.3 is missing, and CHANGELOG.md is out of date

### ISSUE THREAD

**Issue by benma** [rust-bitcoin/rust-bitcoin#3434]
Saw there was new release 0.32.3, but can't find any info on it.

Also the current top line of [CHANGELOG.md](
https://github.com/rust-bitcoin/rust-bitcoin/blob/f3fbd0e0b9dd4f055340b41b7f8a2c31cdecc79d/bitcoin/CHANGELOG.md#0320---todo-add-date-when-we-do-final-release) is:

> 0.32.0 - TODO: Add date when we do final release

But 0.32.0 has been released half a year ago.

Would be great to keep changelogs and releases up to date.

**Comment by tcharding** [comment]
This is fixed in #$3417, thanks for noticing though!

**Comment by tcharding** [comment]
@apoelstra we should make an effort (both of us) to have the PR that updates changelog entry on master for point release ready and mergable at the same time as the release tracking PR , so then you can merge it immediately after tagging and publishing.

**Comment by apoelstra** [comment]
@tcharding agreed. But for this week I think it's ok to do the releases we're trying to do (units and io) and then to update the currently-open #3417.

But once 3417 is in we should be on top of things.

**Comment by tcharding** [comment]
hmm, as far as process goes I think we should get on top of this because some poor soul is bound to see the release as we do it then go looking for release notes? Perhaps we should have a note at the top of the changelog saying "if a recent point release is missing please check the branch, we likely have just not ported the release notes yet e.g., `0.32.x`)

**Comment by apoelstra** [comment]
Another thing we can do is update the link to the CHANGELOG that appears in the README. Currently it points to the "HEAD" branch (which is confusing and wrong, we should change it to "master"). On release branches we should change it to point to the release branches.

And sure, on master we can put a note up saying that if there are missing entries, they likely exist on other branches (and maybe we should ask people to file issues in case we forgot).

**Comment by benma** [comment]
Thanks, I should have checked the open PRs too. 

Best case: CHANGELOG.md is up to date by the time the tag is created.

I don't personally see much value in maintaining https://github.com/rust-bitcoin/rust-bitcoin/releases - a tag, crates.io release and CHANGELOG are sufficient in my opinion. But if you do maintain a releases page, ideally the release there is finalized before a release to crates.io.

I'll close this as you are already on top of it - feel free to keep discussing here or re-opening it if you prefer.

**Comment by benma** [comment]
Similar experience again today. Latest tag for bitcoin is 0.32.5, I wanted to check what's new, but the CHANGELOG stops at 0.32.3. This time I looked for open PRS with "changelog" and couldn't find any open PRs.

**Comment by apoelstra** [comment]
Repoening this issue.

To close it, let's write a release script which

* Takes a package to publish and optionally `--dry-run`
* Checks that the dependencies of the package to publish are already published (since `cargo publish --dry-run`  apparently does not check this).
* Checks that there is a CHANGELOG entry on **both** the branch being published and on `master`.
* Creates a signed tag with a consistent name and message (echoing something so the user knows he needs to touch his yubikey)
* (Maybe) checks that every single other released version of the package has a signed tag and warns if not.
* Runs `cargo publish` and deletes the tag on failure
* Pushes the tag

**Comment by tcharding** [comment]
Yeah I agree. I literally wondered when doing the 0.36.0 PR how we would remember to push the changelog changes to master. I did not think to check the previous ones, face palm.
